lib/configure.js

```diff
@@ -334,6 +341,22 @@ module.exports = function() {
       if (err.message === 'repository not found') {
         process.stdout.write('I didn\'t find your repository in Travis.  Syncing Travis with GitHub…\n');
         return travis.users.sync.post()
+        .catch(function(err) {
+          // Ignore the exception if Travis is already syncing. The status code
+          // of the error is 409 and its message is 'Sync already in progress.
+          // Try again later.', but instead of relying on that we simply perform
+          // an explicit check.
+          return travisIsSyncing().then(function(is_syncing) {
+            if (!is_syncing) {
+              // Check again if it's active on Travis (in case it finished syncing
+              // right before we checked)
+              return ensureActiveInTravis().catch(function() {
+                // Throw the original error.
+                throw err;
+              });
+            }
+          });
+        })
         .then(travisAwaitSyncing)
```

Comment:
I'm more concerned about the complexity of the logic, which makes it harder to hack and can hide bugs. I'd prefer it to be as simple as possible (although no simpler). But I'm ok with the current version.